EIPS/eip-token-standard.md

```diff
@@ -0,0 +1,167 @@
+## Preamble
+
+    EIP: 20
+    Title: ERC-20 Token Standard
+    Author: Fabian Vogelsteller <fabian@ethereum.org>, Vitalik Buterin <vitalik.buterin@ethereum.org>
+    Type: Standard
+    Category: ERC
+    Status: Accepted
+    Created: 2015-11-19
+
+
+## Simple Summary
+
+A standard interface for tokens.
+
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for tokens within smart contracts.
+This standard provides basic functionality to transfer tokens, as well as allow tokens to be approved so they can be spent by another on-chain third party.
+
+
+## Motivation
+
+A standard interface allows any tokens on Ethereum to be re-used by other applications: from wallets to decentralized exchanges.
+
+
+## Specification
+
+## Token
+### Methods
+
+**NOTE**: Callers should handle `false` from `returns (bool success)`.  Callers should not assume that `false` is never returned!
+
+#### name
+
+Returns the name of the token - e.g. `"MyToken"`
+
+``` js
+function name() constant returns (string name)
+```
+
+
+#### symbol
+
+Returns the symbol of the token. E.g. "MYT"
+
+``` js
+function symbol() constant returns (string symbol)
+```
+
+
+
+#### decimals
+
+Returns the number of decimals the token uses - e.g. `8`, means to divide the token amount by `100000000` to get its user representation.
+
+``` js
+function decimals() constant returns (uint8 decimals)
+```
+
+
+#### totalSupply
+
+Returns the total token supply.
+
+``` js
+function totalSupply() constant returns (uint256 totalSupply)
+```
+
+
+
+#### balanceOf
+
+Returns the account balance of another account with address `_owner`.
+
+``` js
+function balanceOf(address _owner) constant returns (uint256 balance)
+```
+
+
+
+#### transfer
+
+Transfers `_value` amount of tokens to address `_to`.
```

Comment:
I agree that transferring zero tokens should usually be treated as success and though should return true, but I'm not sure about event.

According to current specification, "Transfer" event "triggered when tokens are transferred", and one may interpret this as "when tokens change hands".  There are no references to "transfer", "transferFrom" or some other method in event specifications.  So, when "transfer" or "transferFrom" method was called, but no tokens actually changed hands (i.e. zero tokens were transferred or "from" and "to" addresses are the same), no event should probably be logged.  On the other hand, if tokens changed hands for some reason other than execution of "transfer" or "transferFrom" method, the event should be logged.